languageserver/integration/codelenses.go

```diff
@@ -159,72 +257,162 @@ func (i *FlowIntegration) entryPointActions(
+			Range: codelensRange,
+			Command: &protocol.Command{
+				Title:     title,
+				Command:   ClientStartEmulator,
+				Arguments: []interface{}{uri, "Offline"},
+			},
 		}
+		codeLenses = append(codeLenses, codeLens)
+		return codeLenses, nil
+	}
 
-		title = fmt.Sprintf(
-			"submit with account 0x%s",
-			i.activeAddress.Hex(),
+	// If emulator is not up, we need to provide actionless codelens
+	if i.emulatorState == EmulatorStarting {
+		title := fmt.Sprintf(
+			"%s Emulator is starting up. Please wait \u2026",
+			prefixStarting,
 		)
-		argumentListConjunction = "and"
-		command = CommandSubmitTransaction
+		codeLens := &protocol.CodeLens{
+			Range: codelensRange,
+			Command: &protocol.Command{
+				Title: title,
+			},
+		}
+		codeLenses = append(codeLenses, codeLens)
+		return codeLenses, nil
 	}
 
 	argumentLists := entryPointInfo.pragmaArguments[:]
 
 	// If there are no parameters and no pragma argument declarations,
 	// offer execution using no arguments
-	if len(entryPointInfo.parameters) == 0 && len(argumentLists) == 0 {
+	if len(entryPointInfo.parameters) == 0 {
 		argumentLists = append(argumentLists, []cadence.Value{})
 	}
 
-	codeLenses := make([]*protocol.CodeLens, len(argumentLists))
+	signersList := entryPointInfo.pragmaSignersStrings[:]
+	// TODO: resolve check for amount of signers equals to provided by pragma
+	if len(signersList) == 0 {
+		activeAccount := [][]string{[]string{"no-signers"}}
+		signersList = append(signersList, activeAccount...)
+	}
 
-	for i, argumentList := range argumentLists {
-		formattedTitle := title
-		if len(argumentList) > 0 {
-			formattedTitle = fmt.Sprintf(
-				"%s %s %s",
-				title,
-				argumentListConjunction,
-				entryPointInfo.pragmaArgumentStrings[i],
-			)
-		}
+	for index, argumentList := range argumentLists {
+		var title string
 
-		encodedArgumentList := make([]string, len(argumentList))
-		for i, argument := range argumentList {
+		var encodedArgumentList []string
+		for _, argument := range argumentList {
 			encodedArgument, err := jsoncdc.Encode(argument)
 			if err != nil {
 				return nil, err
 			}
-			encodedArgumentList[i] = string(encodedArgument)
+			noSuffix := strings.TrimSuffix(string(encodedArgument), "\n")
+			encodedArgumentList = append(encodedArgumentList, noSuffix)
 		}
 
-		codeLenses[i] = &protocol.CodeLens{
-			Range: conversion.ASTToProtocolRange(
-				position,
-				position,
-			),
-			Command: &protocol.Command{
-				Title:     formattedTitle,
-				Command:   command,
-				Arguments: []interface{}{uri, encodedArgumentList},
-			},
+		switch entryPointInfo.kind {
+
+		case entryPointKindScript:
+			if len(argumentList) > 0 {
+				title = fmt.Sprintf(
+					"%s Execute script with %s",
+					prefixOK,
+					entryPointInfo.pragmaArgumentStrings[index],
+				)
+			} else {
+				title = fmt.Sprintf(
+					"%s Execute script",
+					prefixOK,
+				)
+			}
+
+			argsJSON := convertListToCadenceJSON(encodedArgumentList)
```

Comment:
Instead of preparing the arguments separately and then using string formatting and replacing characters to produce the final JSON (in `convertListToCadenceJSON`), consider JSON marshaling a Go slice with the arguments